t/Privileges.t

```diff
@@ -0,0 +1,55 @@
+use strict;
+use warnings;
+
+use Test;
+use Win32;
+use Config;
+use File::Temp;
+
+plan tests => 7;
+
+ok(ref(Win32::GetProcessPrivileges) eq 'HASH');
+ok(ref(Win32::GetProcessPrivileges(Win32::GetCurrentProcessId())) eq 'HASH');
+
+# All Windows PIDs are divisible by 4. It's an undocumented implementation
+# detail, but it means it's extremely unlikely that the PID below is valid.
+ok(!Win32::GetProcessPrivileges(3423237));
+
+my $whoami = `whoami /priv 2>&1`;
+my $skip = ($? == -1 || $? >> 8) ? '"whoami" command is missing' : 0;
+
+skip($skip, sub{
+    my $privs = Win32::GetProcessPrivileges();
+
+    while ($whoami =~ /^(Se\w+)/mg) {
+        return 0 unless exists $privs->{$1};
+    }
+
+    return 1;
+});
+
+# there isn't really anything to test, we just want to make sure that the
+# function doesn't segfault
+Win32::IsDeveloperModeEnabled();
+ok(1);
+
+Win32::IsSymlinkCreationAllowed();
+ok(1);
+
+$skip = $^O ne 'MSWin32' ? 'windows-only test' : 0;
+$skip ||= !$Config{d_symlink} ? 'this perl doesn\'t have symlink()' : 0;
+
+skip($skip, sub {
+    my $tmpdir = File::Temp->newdir;
+    my $dirname = $tmpdir->dirname;
+
+    if (Win32::IsSymlinkCreationAllowed()) {
+        # we expect success
+        return 0+!!symlink("foo", $tmpdir->dirname . "/new_symlink");
```

Comment:
I find the expression for the return value hard to comprehend:

If the return value is a bool, what is the point of the `0+`. And if it is a number, what is the point of the `0+!!`?

What is wrong with the straightforward expression: `return symlink(...) == 1` (and `return symlink(...) == 0` when you expect failure)?

`symlink` is already defined to return a number:

> Returns 1 for success, 0 otherwise.

Reply:
That kind of normalization, ensuring that the return value is either 0 or 1, was needed in a few projects I worked on, so it's in my muscle memory. But, yeah, I can see how it can be confusing.